torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -25,58 +26,135 @@ static int ceilDiv(const int a, const int b) {
   return (a + b - 1) / b;
 }
 
+// IO data structure for kernel code;
+static auto typeinfo = R"(
+template<typename T, int N>
+struct Tensor {
+  T& operator[](int ind) {
+    return data[ind];
+  };
+
+  T* data;
+  int size[N];
+  int stride[N];
+};
```

Comment:
shouldn't this be `size_t size[N]` and `size_t stride[N]`? I would imagine we'd specialize on fitting in an int later in the future sometime if we want.

Reply:
I can try. I didn't do size_t because it requires a header :/ which becomes tricky with nvrtc....
Ben might have a good solution for that in his jitify though.

Reply:
We use size_t in the generated code all the time, no issues so far.